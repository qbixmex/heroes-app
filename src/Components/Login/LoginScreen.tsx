import { useNavigate } from "react-router-dom";

const LoginScreen = () => {
  const navigate = useNavigate();

  const handleLogin = () => {
    navigate("/", {
      replace: true
    });
  };

  return (
    <>
      <h1 className="text-center display-1 green mt-4">Login Screen</h1>
      <hr />

      <div className="text-center">
        <button
          className="btn btn-outline-primary mt-5"
          onClick={ handleLogin }
        >Login with Google</button>
      </div>
    </>
  );
};

export default LoginScreen;
