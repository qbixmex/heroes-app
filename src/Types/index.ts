export type AuthState = {
  name?: string;
  logged?: boolean;
};

export type AuthAction =
  | { type: types.login, payload: AuthState }
  | { type: types.logout };

export enum types {
  login = "[auth] Login",
  logout = "[auth] Logout"  
}
