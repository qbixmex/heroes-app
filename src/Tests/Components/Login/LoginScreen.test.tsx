import { Route, Routes, MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../../Auth/authContext';
import LoginScreen from '../../../Components/Login/LoginScreen';
import { mount } from 'enzyme';
import { types } from '../../../Types';

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate
}));

// @ts-ignore
describe("Tests on LoginScreen", () => {

  const contextValue = {
    user: {
      logged: false,
    },
    dispatch: jest.fn()
  };

  const wrapper = mount(
    <AuthContext.Provider value={ contextValue }>
      <MemoryRouter initialEntries={ ["/login" ]}>
        <Routes>
          <Route path="/login" element={ <LoginScreen /> } />
        </Routes>
      </MemoryRouter>
    </AuthContext.Provider>
  );

  test("Should match with snapshot", () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test("Should login after click login button", () => {
    const mockEvent = {} as React.MouseEvent<HTMLButtonElement>;

    const handleClick = wrapper.find("button").prop<React.MouseEventHandler>("onClick");

    handleClick( mockEvent );

    expect( contextValue.dispatch ).toHaveBeenCalledWith({
      type: types.login,
      payload: { name: "John Doe" }
    });
    
    expect( mockNavigate ).toHaveBeenCalledWith("/", { replace: true });

    localStorage.setItem("lastPath", "/dc");

    handleClick( mockEvent );

    expect( mockNavigate ).toHaveBeenCalledWith("/dc", { replace: true });
  });

});
