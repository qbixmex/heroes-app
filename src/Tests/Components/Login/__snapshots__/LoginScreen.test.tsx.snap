// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on LoginScreen Should match with snapshot 1`] = `
Array [
  <h1
    className="text-center display-1 green mt-4"
  >
    Login Screen
  </h1>,
  <hr />,
  <div
    className="text-center"
  >
    <button
      className="btn btn-outline-primary mt-5"
      onClick={[Function]}
    >
      Login with Google
    </button>
  </div>,
]
`;
